irispy/new_sji.py

```diff
@@ -0,0 +1,179 @@
+
+    mask : any type, optional
+        Mask for the dataset. Masks should follow the numpy convention
+        that valid data points are marked by False and invalid ones with True.
+        Defaults to None.
+
+    extra_coords : iterable of `tuple`s, each with three entries
+        (`str`, `int`, `astropy.units.quantity` or array-like)
+        Gives the name, axis of data, and values of coordinates of a data axis
+        not included in the WCS object.
+
+    copy : `bool`, optional
+        Indicates whether to save the arguments as copy. True copies every
+        attribute before saving it while False tries to save every parameter
+        as reference. Note however that it is not always possible to save the
+        input as reference.
+        Default is False.
+
+    Examples
+    --------
+    >>> from irispy import sji
+    >>> from irispy.data import sample
+    >>> sji = read_iris_sji_level2_fits(sample.SJI_CUBE_1400)
+    """
+
+    def __repr__(self):
+        return (
+            """
+    SJICube
+    ---------
+    Observatory:\t\t {obs}
+    Instrument:\t\t\t {instrument}
+    Bandpass:\t\t\t {bandpass}
+    Obs. Start:\t\t\t {date_start:{tmf}}
+    Obs. End:\t\t\t {date_end:{tmf}}
+    Instance Start:\t\t {instance_start:{tmf}}
+    Instance End:\t\t {instance_end:{tmf}}
+    Total Frames in Obs.:\t {frame_num}
+    IRIS Obs. id:\t\t {obs_id}
+    IRIS Obs. Description:\t {obs_desc}
+    Cube dimensions:\t\t {dimensions}
+    Axis Types:\t\t\t {axis_types}
+    """.format(obs=self.meta["TELESCOP"],
+               instrument=self.meta["INSTRUME"],
+               bandpass=self.meta["TWAVE1"],
+               date_start=self.meta["DATE_OBS"],
+               date_end=self.meta["DATE_END"],
+               instance_start=self.extra_coords["TIME"]["value"][0],
+               instance_end=self.extra_coords["TIME"]["value"][-1],
+               frame_num=self.meta["NBFRAMES"],
+               obs_id=self.meta["OBSID"],
+               obs_desc=self.meta["OBS_DESC"],
+               axis_types=self.world_axis_physical_types,
+               dimensions=self.dimensions, tmf=TIME_FORMAT))
+
+
+def read_iris_sji_level2_fits(filename):
+    """
+    Read IRIS level 2 SJI FITS from an OBS into an SJICube instance
+
+    Parameters
+    ----------
+    filename : `str`
+        File name to be read
+
+    Returns
+    -------
+    result: 'irispy.sji.SJICube'
+
+    """
+
+    # Open a fits file
+    my_file = fits.open(filename, memmap=True, do_not_scale_image_data=True)
+    # Derive WCS, data and mask for NDCube from fits file.
+    wcs = WCS(my_file[0].header)
+    data = my_file[0].data
+    data_nan_masked = my_file[0].data
+    data_nan_masked[data == BAD_PIXEL_VALUE_UNSCALED] = BAD_PIXEL_VALUE
+    mask = data_nan_masked == BAD_PIXEL_VALUE_UNSCALED
+    # Derive unit and readout noise from detector.
+    exposure_times = my_file[1].data[:, my_file[1].header["EXPTIMES"]]
+    unit = iris_tools.DN_UNIT["SJI"]
```

Comment:
if we have an ```SJICube.scaled``` attribute, this ```unit variable should be defined in an ```if...else...``` statement based on whether ```memmap``` is ```True``` or ```False```.

Reply:
`unit` statement has been added, I am waiting the `DN_UNIT["SJI_UNSCALED"]`